### ISSUE TITLE
macOS link failure - Undefined symbols for architecture x86_64:   "xmain(int, char const**)", referenced from:       -[XWinApplication run] in CocoaMain.o

### ISSUE THREAD

**Issue by SamuelMarks** [alaingalvan/CrossWindow#6]
What am I doing wrong on my macOS 11.1 (20C69)?

### `CMakeLists.txt`
```cmake
cmake_minimum_required(VERSION 3.0)
cmake_policy(SET CMP0048 NEW)
project(pp VERSION 0.0.0 LANGUAGES CXX)

set(DCMAKE_GENERATOR_PLATFORM "x64")
set(CMAKE_CXX_STANDARD 14)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
set(CMAKE_CXX_EXTENSIONS OFF)

set(src src/main.cpp)

add_subdirectory(external/crosswindow)

# ❎ When creating your executable use CrossWindow's abstraction function:
xwin_add_executable(
    # Target
    ${PROJECT_NAME}
    # Source Files (make sure to surround in quotations so CMake treats it as a list)
    "${SOURCE_FILES}"
)

set(libs CrossWindow)

if (APPLE) # or if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
    find_library(COCOA_LIBRARY Cocoa ONLY)
    #target_link_libraries(${PROJECT_NAME} PRIVATE ${COCOA_LIBRARY})
    set(libs ${libs};${COCOA_LIBRARY})
endif()

message("libs ${libs}")
# 🔗 Link CrossWindow to your project:
target_link_libraries(${PROJECT_NAME} ${libs})
```

### `src/main.cpp`
https://github.com/alaingalvan/CrossWindow/blob/b2d4f08/readme.md#usage

### `sh`
```sh
$ git init && mkdir 'external' && cd "$_"
$ git submodule add https://github.com/alaingalvan/crosswindow.git
$ mkdir 'crosswindow/build' && cd "$_"
$ cmake -G 'Xcode' ..
$ cmake --build .
$ cd ../../..
$ tree -L 2
.
├── CMakeLists.txt
├── external
│   └── crosswindow
└── src
    └── main.cpp
$ mkdir build && cd $_
$ cmake -G 'Xcode' ..
-- The CXX compiler identification is AppleClang 12.0.0.12000032
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- The C compiler identification is AppleClang 12.0.0.12000032
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Building CrossWindow for Cocoa
Creating CrossWindow executable:
libs CrossWindow;/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/Cocoa.framework
-- Configuring done
-- Generating done
-- Build files have been written to: temp_dir/cross/build
$ cmake --build .
Command line invocation:
    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -project pp.xcodeproj build -target ALL_BUILD -parallelizeTargets -configuration Debug -hideShellScriptEnvironment

User defaults from command line:
    HideShellScriptEnvironment = YES

note: Using new build system
note: Building targets in parallel
note: Planning build
note: Constructing build description
CreateBuildDirectory temp_dir/cross/build/external/crosswindow (in target 'CrossWindow' from project 'pp')
    cd temp_dir/cross
    builtin-create-build-directory temp_dir/cross/build/external/crosswindow

CreateBuildDirectory temp_dir/cross/build (in target 'pp' from project 'pp')
    cd temp_dir/cross
    builtin-create-build-directory temp_dir/cross/build

WriteAuxiliaryFile temp_dir/cross/build/pp.build/Debug/ZERO_CHECK.build/Script-C15842543262D8CF4EC9B42F.sh (in target 'ZERO_CHECK' from project 'pp')
    cd temp_dir/cross
    write-file temp_dir/cross/build/pp.build/Debug/ZERO_CHECK.build/Script-C15842543262D8CF4EC9B42F.sh

MkDir temp_dir/cross/build/Debug/pp.app (in target 'pp' from project 'pp')
    cd temp_dir/cross
    /bin/mkdir -p temp_dir/cross/build/Debug/pp.app

MkDir temp_dir/cross/build/Debug/pp.app/Contents (in target 'pp' from project 'pp')
    cd temp_dir/cross
    /bin/mkdir -p temp_dir/cross/build/Debug/pp.app/Contents

MkDir temp_dir/cross/build/Debug/pp.app/Contents/MacOS (in target 'pp' from project 'pp')
    cd temp_dir/cross
    /bin/mkdir -p temp_dir/cross/build/Debug/pp.app/Contents/MacOS

PhaseScriptExecution Generate\ CMakeFiles/ZERO_CHECK temp_dir/cross/build/pp.build/Debug/ZERO_CHECK.build/Script-C15842543262D8CF4EC9B42F.sh (in target 'ZERO_CHECK' from project 'pp')
    cd temp_dir/cross
    /bin/sh -c temp_dir/cross/build/pp.build/Debug/ZERO_CHECK.build/Script-C15842543262D8CF4EC9B42F.sh
make: `temp_dir/cross/build/CMakeFiles/cmake.check_cache' is up to date.

CompileC temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Dialogs.o temp_dir/cross/external/crosswindow/src/CrossWindow/Common/Dialogs.cpp normal x86_64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'CrossWindow' from project 'pp')
    cd temp_dir/cross
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c++ -target x86_64-apple-macos11.1 -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-implicit-atomic-properties -Wno-objc-interface-ivars -Wno-arc-repeated-use-of-weak -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wno-undeclared-selector -Wno-deprecated-implementations -Wno-c++11-extensions -DCMAKE_INTDIR\=\"Debug\" -DXWIN_COCOA\=1 -DOBJC_OLD_DISPATCH_PROTOTYPES\=1 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -Winvalid-offsetof -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -Itemp_dir/cross/build/external/crosswindow/Debug/include -Itemp_dir/cross/external/crosswindow/src -Itemp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources-normal/x86_64 -Itemp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources/x86_64 -Itemp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources -Ftemp_dir/cross/build/external/crosswindow/Debug -std\=c++14 -x objective-c++ -MMD -MT dependencies -MF temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Dialogs.d --serialize-diagnostics temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Dialogs.dia -c temp_dir/cross/external/crosswindow/src/CrossWindow/Common/Dialogs.cpp -o temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Dialogs.o

CompileC temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Init.o temp_dir/cross/external/crosswindow/src/CrossWindow/Common/Init.cpp normal x86_64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'CrossWindow' from project 'pp')
    cd temp_dir/cross
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c++ -target x86_64-apple-macos11.1 -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-implicit-atomic-properties -Wno-objc-interface-ivars -Wno-arc-repeated-use-of-weak -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wno-undeclared-selector -Wno-deprecated-implementations -Wno-c++11-extensions -DCMAKE_INTDIR\=\"Debug\" -DXWIN_COCOA\=1 -DOBJC_OLD_DISPATCH_PROTOTYPES\=1 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -Winvalid-offsetof -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -Itemp_dir/cross/build/external/crosswindow/Debug/include -Itemp_dir/cross/external/crosswindow/src -Itemp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources-normal/x86_64 -Itemp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources/x86_64 -Itemp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources -Ftemp_dir/cross/build/external/crosswindow/Debug -std\=c++14 -x objective-c++ -MMD -MT dependencies -MF temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Init.d --serialize-diagnostics temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Init.dia -c temp_dir/cross/external/crosswindow/src/CrossWindow/Common/Init.cpp -o temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Init.o

WriteAuxiliaryFile temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/CrossWindow.LinkFileList (in target 'CrossWindow' from project 'pp')
    cd temp_dir/cross
    write-file temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/CrossWindow.LinkFileList

CompileC temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/WindowDesc.o temp_dir/cross/external/crosswindow/src/CrossWindow/Common/WindowDesc.cpp normal x86_64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'CrossWindow' from project 'pp')
    cd temp_dir/cross
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c++ -target x86_64-apple-macos11.1 -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-implicit-atomic-properties -Wno-objc-interface-ivars -Wno-arc-repeated-use-of-weak -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wno-undeclared-selector -Wno-deprecated-implementations -Wno-c++11-extensions -DCMAKE_INTDIR\=\"Debug\" -DXWIN_COCOA\=1 -DOBJC_OLD_DISPATCH_PROTOTYPES\=1 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -Winvalid-offsetof -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -Itemp_dir/cross/build/external/crosswindow/Debug/include -Itemp_dir/cross/external/crosswindow/src -Itemp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources-normal/x86_64 -Itemp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources/x86_64 -Itemp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources -Ftemp_dir/cross/build/external/crosswindow/Debug -std\=c++14 -x objective-c++ -MMD -MT dependencies -MF temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/WindowDesc.d --serialize-diagnostics temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/WindowDesc.dia -c temp_dir/cross/external/crosswindow/src/CrossWindow/Common/WindowDesc.cpp -o temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/WindowDesc.o

CompileC temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/CocoaWindow.o temp_dir/cross/external/crosswindow/src/CrossWindow/Cocoa/CocoaWindow.mm normal x86_64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'CrossWindow' from project 'pp')
    cd temp_dir/cross
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c++ -target x86_64-apple-macos11.1 -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-implicit-atomic-properties -Wno-objc-interface-ivars -Wno-arc-repeated-use-of-weak -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wno-undeclared-selector -Wno-deprecated-implementations -Wno-c++11-extensions -DCMAKE_INTDIR\=\"Debug\" -DXWIN_COCOA\=1 -DOBJC_OLD_DISPATCH_PROTOTYPES\=1 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -Winvalid-offsetof -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -Itemp_dir/cross/build/external/crosswindow/Debug/include -Itemp_dir/cross/external/crosswindow/src -Itemp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources-normal/x86_64 -Itemp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources/x86_64 -Itemp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources -Ftemp_dir/cross/build/external/crosswindow/Debug -std\=c++14 -x objective-c++ -MMD -MT dependencies -MF temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/CocoaWindow.d --serialize-diagnostics temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/CocoaWindow.dia -c temp_dir/cross/external/crosswindow/src/CrossWindow/Cocoa/CocoaWindow.mm -o temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/CocoaWindow.o
temp_dir/cross/external/crosswindow/src/CrossWindow/Cocoa/CocoaWindow.mm:190:13: warning: 'CAOpenGLLayer' is deprecated: first deprecated in macOS 10.14 - OpenGL is deprecated [-Wdeprecated-declarations]
                layer = [[CAOpenGLLayer alloc] init];
                          ^
In file included from temp_dir/cross/external/crosswindow/src/CrossWindow/Cocoa/CocoaWindow.mm:2:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:13:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/AppKit.framework/Headers/AppKit.h:198:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSOpenGLLayer.h:11:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/QuartzCore.framework/Headers/CAOpenGLLayer.h:17:12: note: 'CAOpenGLLayer' has been explicitly marked deprecated here
@interface CAOpenGLLayer : CALayer
           ^
temp_dir/cross/external/crosswindow/src/CrossWindow/Cocoa/CocoaWindow.mm:191:30: warning: 'CAOpenGLLayer' is deprecated: first deprecated in macOS 10.14 - OpenGL is deprecated [-Wdeprecated-declarations]
                [(XWinView*)view setLayer:(CAOpenGLLayer*)layer];
                                           ^
In file included from temp_dir/cross/external/crosswindow/src/CrossWindow/Cocoa/CocoaWindow.mm:2:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:13:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/AppKit.framework/Headers/AppKit.h:198:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSOpenGLLayer.h:11:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/QuartzCore.framework/Headers/CAOpenGLLayer.h:17:12: note: 'CAOpenGLLayer' has been explicitly marked deprecated here
@interface CAOpenGLLayer : CALayer
           ^
temp_dir/cross/external/crosswindow/src/CrossWindow/Cocoa/CocoaWindow.mm:192:3: warning: 'CAOpenGLLayer' is deprecated: first deprecated in macOS 10.14 - OpenGL is deprecated [-Wdeprecated-declarations]
                CAOpenGLLayer* l = ((CAOpenGLLayer*)layer);
                ^
In file included from temp_dir/cross/external/crosswindow/src/CrossWindow/Cocoa/CocoaWindow.mm:2:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:13:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/AppKit.framework/Headers/AppKit.h:198:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSOpenGLLayer.h:11:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/QuartzCore.framework/Headers/CAOpenGLLayer.h:17:12: note: 'CAOpenGLLayer' has been explicitly marked deprecated here
@interface CAOpenGLLayer : CALayer
           ^
temp_dir/cross/external/crosswindow/src/CrossWindow/Cocoa/CocoaWindow.mm:192:24: warning: 'CAOpenGLLayer' is deprecated: first deprecated in macOS 10.14 - OpenGL is deprecated [-Wdeprecated-declarations]
                CAOpenGLLayer* l = ((CAOpenGLLayer*)layer);
                                     ^
In file included from temp_dir/cross/external/crosswindow/src/CrossWindow/Cocoa/CocoaWindow.mm:2:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:13:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/AppKit.framework/Headers/AppKit.h:198:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSOpenGLLayer.h:11:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/QuartzCore.framework/Headers/CAOpenGLLayer.h:17:12: note: 'CAOpenGLLayer' has been explicitly marked deprecated here
@interface CAOpenGLLayer : CALayer
           ^
4 warnings generated.

CompileC temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/CocoaEventQueue.o temp_dir/cross/external/crosswindow/src/CrossWindow/Cocoa/CocoaEventQueue.mm normal x86_64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'CrossWindow' from project 'pp')
    cd temp_dir/cross
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c++ -target x86_64-apple-macos11.1 -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-implicit-atomic-properties -Wno-objc-interface-ivars -Wno-arc-repeated-use-of-weak -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wno-undeclared-selector -Wno-deprecated-implementations -Wno-c++11-extensions -DCMAKE_INTDIR\=\"Debug\" -DXWIN_COCOA\=1 -DOBJC_OLD_DISPATCH_PROTOTYPES\=1 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -Winvalid-offsetof -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -Itemp_dir/cross/build/external/crosswindow/Debug/include -Itemp_dir/cross/external/crosswindow/src -Itemp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources-normal/x86_64 -Itemp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources/x86_64 -Itemp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources -Ftemp_dir/cross/build/external/crosswindow/Debug -std\=c++14 -x objective-c++ -MMD -MT dependencies -MF temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/CocoaEventQueue.d --serialize-diagnostics temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/CocoaEventQueue.dia -c temp_dir/cross/external/crosswindow/src/CrossWindow/Cocoa/CocoaEventQueue.mm -o temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/CocoaEventQueue.o

CompileC temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Event.o temp_dir/cross/external/crosswindow/src/CrossWindow/Common/Event.cpp normal x86_64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'CrossWindow' from project 'pp')
    cd temp_dir/cross
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c++ -target x86_64-apple-macos11.1 -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-implicit-atomic-properties -Wno-objc-interface-ivars -Wno-arc-repeated-use-of-weak -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wno-undeclared-selector -Wno-deprecated-implementations -Wno-c++11-extensions -DCMAKE_INTDIR\=\"Debug\" -DXWIN_COCOA\=1 -DOBJC_OLD_DISPATCH_PROTOTYPES\=1 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -Winvalid-offsetof -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -Itemp_dir/cross/build/external/crosswindow/Debug/include -Itemp_dir/cross/external/crosswindow/src -Itemp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources-normal/x86_64 -Itemp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources/x86_64 -Itemp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources -Ftemp_dir/cross/build/external/crosswindow/Debug -std\=c++14 -x objective-c++ -MMD -MT dependencies -MF temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Event.d --serialize-diagnostics temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Event.dia -c temp_dir/cross/external/crosswindow/src/CrossWindow/Common/Event.cpp -o temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Event.o

WriteAuxiliaryFile temp_dir/cross/build/pp.build/Debug/pp.build/DerivedSources/Entitlements.plist (in target 'pp' from project 'pp')
    cd temp_dir/cross
    write-file temp_dir/cross/build/pp.build/Debug/pp.build/DerivedSources/Entitlements.plist

ProcessProductPackaging "" temp_dir/cross/build/pp.build/Debug/pp.build/pp.app.xcent (in target 'pp' from project 'pp')
    cd temp_dir/cross
    

Entitlements:

{
    "com.apple.security.get-task-allow" = 1;
}


    builtin-productPackagingUtility -entitlements -format xml -o temp_dir/cross/build/pp.build/Debug/pp.build/pp.app.xcent

CompileC temp_dir/cross/build/pp.build/Debug/pp.build/Objects-normal/x86_64/CocoaMain.o temp_dir/cross/external/crosswindow/src/CrossWindow/Main/CocoaMain.mm normal x86_64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'pp' from project 'pp')
    cd temp_dir/cross
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c++ -target x86_64-apple-macos11.1 -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-implicit-atomic-properties -Wno-objc-interface-ivars -Wno-arc-repeated-use-of-weak -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wno-undeclared-selector -Wno-deprecated-implementations -Wno-c++11-extensions -DCMAKE_INTDIR\=\"Debug\" -DXWIN_COCOA\=1 -DOBJC_OLD_DISPATCH_PROTOTYPES\=1 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -Winvalid-offsetof -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -Itemp_dir/cross/build/Debug/include -Itemp_dir/cross/external/crosswindow/src -Itemp_dir/cross/build/pp.build/Debug/pp.build/DerivedSources-normal/x86_64 -Itemp_dir/cross/build/pp.build/Debug/pp.build/DerivedSources/x86_64 -Itemp_dir/cross/build/pp.build/Debug/pp.build/DerivedSources -Ftemp_dir/cross/build/Debug -std\=c++14 -MMD -MT dependencies -MF temp_dir/cross/build/pp.build/Debug/pp.build/Objects-normal/x86_64/CocoaMain.d --serialize-diagnostics temp_dir/cross/build/pp.build/Debug/pp.build/Objects-normal/x86_64/CocoaMain.dia -c temp_dir/cross/external/crosswindow/src/CrossWindow/Main/CocoaMain.mm -o temp_dir/cross/build/pp.build/Debug/pp.build/Objects-normal/x86_64/CocoaMain.o

Libtool temp_dir/cross/build/external/crosswindow/Debug/libCrossWindow.a normal (in target 'CrossWindow' from project 'pp')
    cd temp_dir/cross
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool -static -arch_only x86_64 -D -syslibroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -Ltemp_dir/cross/build/external/crosswindow/Debug -filelist temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/CrossWindow.LinkFileList -dependency_info temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/CrossWindow_libtool_dependency_info.dat -o temp_dir/cross/build/external/crosswindow/Debug/libCrossWindow.a
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: file: temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Dialogs.o has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: file: temp_dir/cross/build/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/WindowDesc.o has no symbols

WriteAuxiliaryFile temp_dir/cross/build/pp.build/Debug/pp.build/Objects-normal/x86_64/pp.LinkFileList (in target 'pp' from project 'pp')
    cd temp_dir/cross
    write-file temp_dir/cross/build/pp.build/Debug/pp.build/Objects-normal/x86_64/pp.LinkFileList

ProcessInfoPlistFile temp_dir/cross/build/Debug/pp.app/Contents/Info.plist temp_dir/cross/build/CMakeFiles/pp.dir/Info.plist (in target 'pp' from project 'pp')
    cd temp_dir/cross
    builtin-infoPlistUtility temp_dir/cross/build/CMakeFiles/pp.dir/Info.plist -producttype com.apple.product-type.application -genpkginfo temp_dir/cross/build/Debug/pp.app/Contents/PkgInfo -expandbuildsettings -platform macosx -o temp_dir/cross/build/Debug/pp.app/Contents/Info.plist

Ld temp_dir/cross/build/Debug/pp.app/Contents/MacOS/pp normal (in target 'pp' from project 'pp')
    cd temp_dir/cross
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -target x86_64-apple-macos11.1 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -Ltemp_dir/cross/build/Debug -Ftemp_dir/cross/build/Debug -filelist temp_dir/cross/build/pp.build/Debug/pp.build/Objects-normal/x86_64/pp.LinkFileList -Xlinker -object_path_lto -Xlinker temp_dir/cross/build/pp.build/Debug/pp.build/Objects-normal/x86_64/pp_lto.o -Xlinker -no_deduplicate -fobjc-link-runtime -Wl,-search_paths_first -Wl,-headerpad_max_install_names temp_dir/cross/build/external/crosswindow/Debug/libCrossWindow.a -framework Cocoa -framework QuartzCore -Xlinker -no_adhoc_codesign -Xlinker -dependency_info -Xlinker temp_dir/cross/build/pp.build/Debug/pp.build/Objects-normal/x86_64/pp_dependency_info.dat -o temp_dir/cross/build/Debug/pp.app/Contents/MacOS/pp
Undefined symbols for architecture x86_64:
  "xmain(int, char const**)", referenced from:
      -[XWinApplication run] in CocoaMain.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

** BUILD FAILED **


The following build commands failed:
	Ld temp_dir/cross/build/Debug/pp.app/Contents/MacOS/pp normal
(1 failure)
```

**Comment by alaingalvan** [comment]
I did notice in your CMakeLists.txt file that you didn't pass any source variables to `xwin_create_executable`:

```cmake
# 👇 So here's your sources:
set(src src/main.cpp)

add_subdirectory(external/crosswindow)

# ❌ But they're not passed into the 3rd argument here:
xwin_add_executable(
    # Target
    ${PROJECT_NAME}
    # Source Files (make sure to surround in quotations so CMake treats it as a list)
    "${SOURCE_FILES}"
)
```

Could you try replacing `SOURCE_FILES` with the variable `src` you made there? 

Also, we have a few examples you can try out over [here](https://github.com/alaingalvan/CrossWindow-Demos/tree/master/src/04-cross-platform-hello-triangle). I'd recommend playing with that one if you want to work with Metal or OpenGL in MacOS. Keep me posted if anything.

**Comment by SamuelMarks** [comment]
Oh silly me. Woth correct `set` of src files, I now get:
```sh
Command line invocation:
    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -project pp.xcodeproj build -target ALL_BUILD -parallelizeTargets -configuration Debug -hideShellScriptEnvironment

User defaults from command line:
    HideShellScriptEnvironment = YES

note: Using new build system
note: Building targets in parallel
note: Planning build
note: Constructing build description
CreateBuildDirectory temp_dir/cross/cmake-build-debug (in target 'pp' from project 'pp')
    cd temp_dir/cross
    builtin-create-build-directory temp_dir/cross/cmake-build-debug

CreateBuildDirectory temp_dir/cross/cmake-build-debug/external/crosswindow (in target 'CrossWindow' from project 'pp')
    cd temp_dir/cross
    builtin-create-build-directory temp_dir/cross/cmake-build-debug/external/crosswindow

WriteAuxiliaryFile temp_dir/cross/cmake-build-debug/pp.build/Debug/ZERO_CHECK.build/Script-3FCA2972E4C496F8E995BCF4.sh (in target 'ZERO_CHECK' from project 'pp')
    cd temp_dir/cross
    write-file temp_dir/cross/cmake-build-debug/pp.build/Debug/ZERO_CHECK.build/Script-3FCA2972E4C496F8E995BCF4.sh

MkDir temp_dir/cross/cmake-build-debug/Debug/pp.app (in target 'pp' from project 'pp')
    cd temp_dir/cross
    /bin/mkdir -p temp_dir/cross/cmake-build-debug/Debug/pp.app

MkDir temp_dir/cross/cmake-build-debug/Debug/pp.app/Contents (in target 'pp' from project 'pp')
    cd temp_dir/cross
    /bin/mkdir -p temp_dir/cross/cmake-build-debug/Debug/pp.app/Contents

MkDir temp_dir/cross/cmake-build-debug/Debug/pp.app/Contents/MacOS (in target 'pp' from project 'pp')
    cd temp_dir/cross
    /bin/mkdir -p temp_dir/cross/cmake-build-debug/Debug/pp.app/Contents/MacOS

CompileC temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Init.o temp_dir/cross/external/crosswindow/src/CrossWindow/Common/Init.cpp normal x86_64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'CrossWindow' from project 'pp')
    cd temp_dir/cross
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c++ -target x86_64-apple-macos11.1 -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-implicit-atomic-properties -Wno-objc-interface-ivars -Wno-arc-repeated-use-of-weak -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wno-undeclared-selector -Wno-deprecated-implementations -Wno-c++11-extensions -DCMAKE_INTDIR\=\"Debug\" -DXWIN_COCOA\=1 -DOBJC_OLD_DISPATCH_PROTOTYPES\=1 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -Winvalid-offsetof -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -Itemp_dir/cross/cmake-build-debug/external/crosswindow/Debug/include -Itemp_dir/cross/external/crosswindow/src -Itemp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources-normal/x86_64 -Itemp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources/x86_64 -Itemp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources -Ftemp_dir/cross/cmake-build-debug/external/crosswindow/Debug -std\=c++14 -x objective-c++ -MMD -MT dependencies -MF temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Init.d --serialize-diagnostics temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Init.dia -c temp_dir/cross/external/crosswindow/src/CrossWindow/Common/Init.cpp -o temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Init.o

WriteAuxiliaryFile temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/CrossWindow.LinkFileList (in target 'CrossWindow' from project 'pp')
    cd temp_dir/cross
    write-file temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/CrossWindow.LinkFileList

CompileC temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/CocoaEventQueue.o temp_dir/cross/external/crosswindow/src/CrossWindow/Cocoa/CocoaEventQueue.mm normal x86_64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'CrossWindow' from project 'pp')
    cd temp_dir/cross
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c++ -target x86_64-apple-macos11.1 -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-implicit-atomic-properties -Wno-objc-interface-ivars -Wno-arc-repeated-use-of-weak -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wno-undeclared-selector -Wno-deprecated-implementations -Wno-c++11-extensions -DCMAKE_INTDIR\=\"Debug\" -DXWIN_COCOA\=1 -DOBJC_OLD_DISPATCH_PROTOTYPES\=1 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -Winvalid-offsetof -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -Itemp_dir/cross/cmake-build-debug/external/crosswindow/Debug/include -Itemp_dir/cross/external/crosswindow/src -Itemp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources-normal/x86_64 -Itemp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources/x86_64 -Itemp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources -Ftemp_dir/cross/cmake-build-debug/external/crosswindow/Debug -std\=c++14 -x objective-c++ -MMD -MT dependencies -MF temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/CocoaEventQueue.d --serialize-diagnostics temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/CocoaEventQueue.dia -c temp_dir/cross/external/crosswindow/src/CrossWindow/Cocoa/CocoaEventQueue.mm -o temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/CocoaEventQueue.o

CompileC temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Dialogs.o temp_dir/cross/external/crosswindow/src/CrossWindow/Common/Dialogs.cpp normal x86_64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'CrossWindow' from project 'pp')
    cd temp_dir/cross
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c++ -target x86_64-apple-macos11.1 -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-implicit-atomic-properties -Wno-objc-interface-ivars -Wno-arc-repeated-use-of-weak -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wno-undeclared-selector -Wno-deprecated-implementations -Wno-c++11-extensions -DCMAKE_INTDIR\=\"Debug\" -DXWIN_COCOA\=1 -DOBJC_OLD_DISPATCH_PROTOTYPES\=1 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -Winvalid-offsetof -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -Itemp_dir/cross/cmake-build-debug/external/crosswindow/Debug/include -Itemp_dir/cross/external/crosswindow/src -Itemp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources-normal/x86_64 -Itemp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources/x86_64 -Itemp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources -Ftemp_dir/cross/cmake-build-debug/external/crosswindow/Debug -std\=c++14 -x objective-c++ -MMD -MT dependencies -MF temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Dialogs.d --serialize-diagnostics temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Dialogs.dia -c temp_dir/cross/external/crosswindow/src/CrossWindow/Common/Dialogs.cpp -o temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Dialogs.o

CompileC temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/CocoaWindow.o temp_dir/cross/external/crosswindow/src/CrossWindow/Cocoa/CocoaWindow.mm normal x86_64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'CrossWindow' from project 'pp')
    cd temp_dir/cross
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c++ -target x86_64-apple-macos11.1 -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-implicit-atomic-properties -Wno-objc-interface-ivars -Wno-arc-repeated-use-of-weak -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wno-undeclared-selector -Wno-deprecated-implementations -Wno-c++11-extensions -DCMAKE_INTDIR\=\"Debug\" -DXWIN_COCOA\=1 -DOBJC_OLD_DISPATCH_PROTOTYPES\=1 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -Winvalid-offsetof -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -Itemp_dir/cross/cmake-build-debug/external/crosswindow/Debug/include -Itemp_dir/cross/external/crosswindow/src -Itemp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources-normal/x86_64 -Itemp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources/x86_64 -Itemp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources -Ftemp_dir/cross/cmake-build-debug/external/crosswindow/Debug -std\=c++14 -x objective-c++ -MMD -MT dependencies -MF temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/CocoaWindow.d --serialize-diagnostics temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/CocoaWindow.dia -c temp_dir/cross/external/crosswindow/src/CrossWindow/Cocoa/CocoaWindow.mm -o temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/CocoaWindow.o

PhaseScriptExecution Generate\ CMakeFiles/ZERO_CHECK temp_dir/cross/cmake-build-debug/pp.build/Debug/ZERO_CHECK.build/Script-3FCA2972E4C496F8E995BCF4.sh (in target 'ZERO_CHECK' from project 'pp')
    cd temp_dir/cross
    /bin/sh -c temp_dir/cross/cmake-build-debug/pp.build/Debug/ZERO_CHECK.build/Script-3FCA2972E4C496F8E995BCF4.sh
make: `temp_dir/cross/cmake-build-debug/CMakeFiles/cmake.check_cache' is up to date.

CompileC temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/WindowDesc.o temp_dir/cross/external/crosswindow/src/CrossWindow/Common/WindowDesc.cpp normal x86_64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'CrossWindow' from project 'pp')
    cd temp_dir/cross
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c++ -target x86_64-apple-macos11.1 -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-implicit-atomic-properties -Wno-objc-interface-ivars -Wno-arc-repeated-use-of-weak -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wno-undeclared-selector -Wno-deprecated-implementations -Wno-c++11-extensions -DCMAKE_INTDIR\=\"Debug\" -DXWIN_COCOA\=1 -DOBJC_OLD_DISPATCH_PROTOTYPES\=1 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -Winvalid-offsetof -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -Itemp_dir/cross/cmake-build-debug/external/crosswindow/Debug/include -Itemp_dir/cross/external/crosswindow/src -Itemp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources-normal/x86_64 -Itemp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources/x86_64 -Itemp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources -Ftemp_dir/cross/cmake-build-debug/external/crosswindow/Debug -std\=c++14 -x objective-c++ -MMD -MT dependencies -MF temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/WindowDesc.d --serialize-diagnostics temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/WindowDesc.dia -c temp_dir/cross/external/crosswindow/src/CrossWindow/Common/WindowDesc.cpp -o temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/WindowDesc.o

CompileC temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Event.o temp_dir/cross/external/crosswindow/src/CrossWindow/Common/Event.cpp normal x86_64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'CrossWindow' from project 'pp')
    cd temp_dir/cross
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c++ -target x86_64-apple-macos11.1 -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-implicit-atomic-properties -Wno-objc-interface-ivars -Wno-arc-repeated-use-of-weak -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wno-undeclared-selector -Wno-deprecated-implementations -Wno-c++11-extensions -DCMAKE_INTDIR\=\"Debug\" -DXWIN_COCOA\=1 -DOBJC_OLD_DISPATCH_PROTOTYPES\=1 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -Winvalid-offsetof -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -Itemp_dir/cross/cmake-build-debug/external/crosswindow/Debug/include -Itemp_dir/cross/external/crosswindow/src -Itemp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources-normal/x86_64 -Itemp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources/x86_64 -Itemp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/DerivedSources -Ftemp_dir/cross/cmake-build-debug/external/crosswindow/Debug -std\=c++14 -x objective-c++ -MMD -MT dependencies -MF temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Event.d --serialize-diagnostics temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Event.dia -c temp_dir/cross/external/crosswindow/src/CrossWindow/Common/Event.cpp -o temp_dir/cross/cmake-build-debug/external/crosswindow/pp.build/Debug/CrossWindow.build/Objects-normal/x86_64/Event.o

WriteAuxiliaryFile temp_dir/cross/cmake-build-debug/pp.build/Debug/pp.build/DerivedSources/Entitlements.plist (in target 'pp' from project 'pp')
    cd temp_dir/cross
    write-file temp_dir/cross/cmake-build-debug/pp.build/Debug/pp.build/DerivedSources/Entitlements.plist

ProcessProductPackaging "" temp_dir/cross/cmake-build-debug/pp.build/Debug/pp.build/pp.app.xcent (in target 'pp' from project 'pp')
    cd temp_dir/cross
    

Entitlements:

{
    "com.apple.security.get-task-allow" = 1;
}


    builtin-productPackagingUtility -entitlements -format xml -o temp_dir/cross/cmake-build-debug/pp.build/Debug/pp.build/pp.app.xcent

CompileC temp_dir/cross/cmake-build-debug/pp.build/Debug/pp.build/Objects-normal/x86_64/CocoaMain.o temp_dir/cross/external/crosswindow/src/CrossWindow/Main/CocoaMain.mm normal x86_64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'pp' from project 'pp')
    cd temp_dir/cross
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c++ -target x86_64-apple-macos11.1 -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-implicit-atomic-properties -Wno-objc-interface-ivars -Wno-arc-repeated-use-of-weak -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wno-undeclared-selector -Wno-deprecated-implementations -Wno-c++11-extensions -DCMAKE_INTDIR\=\"Debug\" -DXWIN_COCOA\=1 -DOBJC_OLD_DISPATCH_PROTOTYPES\=1 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -Winvalid-offsetof -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -Itemp_dir/cross/cmake-build-debug/Debug/include -Itemp_dir/cross/external/crosswindow/src -Itemp_dir/cross/cmake-build-debug/pp.build/Debug/pp.build/DerivedSources-normal/x86_64 -Itemp_dir/cross/cmake-build-debug/pp.build/Debug/pp.build/DerivedSources/x86_64 -Itemp_dir/cross/cmake-build-debug/pp.build/Debug/pp.build/DerivedSources -Ftemp_dir/cross/cmake-build-debug/Debug -std\=c++14 -MMD -MT dependencies -MF temp_dir/cross/cmake-build-debug/pp.build/Debug/pp.build/Objects-normal/x86_64/CocoaMain.d --serialize-diagnostics temp_dir/cross/cmake-build-debug/pp.build/Debug/pp.build/Objects-normal/x86_64/CocoaMain.dia -c temp_dir/cross/external/crosswindow/src/CrossWindow/Main/CocoaMain.mm -o temp_dir/cross/cmake-build-debug/pp.build/Debug/pp.build/Objects-normal/x86_64/CocoaMain.o

CompileC temp_dir/cross/cmake-build-debug/pp.build/Debug/pp.build/Objects-normal/x86_64/main.o temp_dir/cross/src/main.cpp normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'pp' from project 'pp')
    cd temp_dir/cross
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x c++ -target x86_64-apple-macos11.1 -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-c++11-extensions -DCMAKE_INTDIR\=\"Debug\" -DXWIN_COCOA\=1 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -fasm-blocks -fstrict-aliasing -Wdeprecated-declarations -Winvalid-offsetof -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -Itemp_dir/cross/cmake-build-debug/Debug/include -Itemp_dir/cross/external/crosswindow/src -Itemp_dir/cross/cmake-build-debug/pp.build/Debug/pp.build/DerivedSources-normal/x86_64 -Itemp_dir/cross/cmake-build-debug/pp.build/Debug/pp.build/DerivedSources/x86_64 -Itemp_dir/cross/cmake-build-debug/pp.build/Debug/pp.build/DerivedSources -Ftemp_dir/cross/cmake-build-debug/Debug -std\=c++14 -MMD -MT dependencies -MF temp_dir/cross/cmake-build-debug/pp.build/Debug/pp.build/Objects-normal/x86_64/main.d --serialize-diagnostics temp_dir/cross/cmake-build-debug/pp.build/Debug/pp.build/Objects-normal/x86_64/main.dia -c temp_dir/cross/src/main.cpp -o temp_dir/cross/cmake-build-debug/pp.build/Debug/pp.build/Objects-normal/x86_64/main.o
temp_dir/cross/src/main.cpp:35:48: error: no member named 'Mouse' in 'xwin::EventType'
            if (event.type == xwin::EventType::Mouse)
                              ~~~~~~~~~~~~~~~~~^
temp_dir/cross/src/main.cpp:37:29: error: no type named 'MouseData' in namespace 'xwin'
                const xwin::MouseData mouse = event.data.mouse;
                      ~~~~~~^
temp_dir/cross/src/main.cpp:37:58: error: no member named 'mouse' in 'xwin::EventData'
                const xwin::MouseData mouse = event.data.mouse;
                                              ~~~~~~~~~~ ^
3 errors generated.

** BUILD FAILED **


The following build commands failed:
	CompileC temp_dir/cross/cmake-build-debug/pp.build/Debug/pp.build/Objects-normal/x86_64/main.o temp_dir/cross/src/main.cpp normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)
```

I'll try one of your other examples later. In my current iteration I just need forms, trees (nice-to-have) and progress bars. So nothing fancy.

**Comment by alaingalvan** [comment]
Oh thanks, that looks like an error on our end, I'm working on a patch to fix that and will edit this post when it's done.
